cmd/resolutioncli/client.go

```diff
@@ -20,16 +20,22 @@ import (
 	"context"
 
 	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
 
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata/client"
+	"github.com/operator-framework/operator-controller/internal/controllers"
 )
 
 type indexRefClient struct {
-	renderer     action.Render
-	bundlesCache []*catalogmetadata.Bundle
+	renderer      action.Render
+	bundlesCache  []*catalogmetadata.Bundle
+	channelsCache []*catalogmetadata.Channel
+	packagesCache []*catalogmetadata.Package
 }
 
+var _ controllers.CatalogProvider = &indexRefClient{}
```

Comment:
super nit - this allocates for an `indexRefClient`. If you instead write:

```go
var _ controllers.CatalogProvider = (*indexRefClient)(nil)
```

you're casting a `nil` pointer which gets you the type-checking you want at no runtime cost

Reply:
TIL 😁